src/React.re

```diff
@@ -22,13 +22,14 @@ external createElementVariadic:
   "createElement";
 
 module Ref = {
-  type t('value);
+  type t('value) = {mutable current: Js.nullable('value)};
```

Comment:
I don't think this will work. First of all it will break all existing code that does

```reason
let myRef = React.useRef(Js.Nullable.null);

<SomeComponent ref=myRef />
```

because the type of `myRef` will then be `Js.nullable(Js.nullable('value))`.

Second, refs not referring to a React component need not be nullable, e.g. when you do

```reason
let myRef = React.useRef("my initial string value");
```

So I would propose to keep the definition from my PR and add a module `ComponentRef` with a separate type plus your helper functions:

```reason
type ref('value) = {mutable current: 'value};

module Ref = {
   // as in my PR for compatibility
};

module ComponentRef = {
  type t('value) = ref(Js.nullable('value));

  // plus your helper functions
}
```

Reply:
I'm aware it's a breaking change which is why I did not bring this up earlier. I already had in mind a second type like that, but I also didn't want to be stuck with `React.ComponentRef.t` (which might mean another migration when `React.Ref` disappears) or `React.Ref.componentRef`. That's why your PR simplifies things, at least, IMO. If that is not desired, I'll revise to add a second type or module. 

Once your PR is merged and `useRef` returns `ref('a)`, however, your example will not break. Otherwise, `createRef` would serve for components and `useRef` for everything else.